test_conformance/extensions/cl_khr_semaphore/test_semaphores_negative_create.cpp

```diff
@@ -0,0 +1,672 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "semaphore_base.h"
+
+#include "harness/errorHelpers.h"
+#include <chrono>
+#include <system_error>
+#include <thread>
+#include <vector>
+
+namespace {
+
+// CL_INVALID_CONTEXT if context is nullptr.
+struct CreateInvalidContext : public SemaphoreTestBase
+{
+    CreateInvalidContext(cl_device_id device, cl_context context,
+                         cl_command_queue queue)
+        : SemaphoreTestBase(device, context, queue)
+    {}
+
+    cl_int Run() override
+    {
+        // Create semaphore
+        cl_semaphore_properties_khr sema_props[] = {
+            (cl_semaphore_properties_khr)CL_SEMAPHORE_TYPE_KHR,
+            (cl_semaphore_properties_khr)CL_SEMAPHORE_TYPE_BINARY_KHR, 0
+        };
+
+        cl_int err = CL_SUCCESS;
+        semaphore =
+            clCreateSemaphoreWithPropertiesKHR(nullptr, sema_props, &err);
+        test_failure_error(
+            err, CL_INVALID_CONTEXT,
+            "Unexpected clCreateSemaphoreWithPropertiesKHR return");
+
+        return CL_SUCCESS;
+    }
+};
+
+// scope guard helper to ensure proper releasing of sub devices
+struct SubDevicesScopeGuarded
+{
+    SubDevicesScopeGuarded(const cl_int dev_count)
+    {
+        sub_devices.resize(dev_count);
+    }
+    ~SubDevicesScopeGuarded()
+    {
+        for (auto& device : sub_devices)
+        {
+            cl_int err = clReleaseDevice(device);
+            if (err != CL_SUCCESS)
+                log_error("\n Releasing sub-device failed \n");
+        }
+    }
+
+    std::vector<cl_device_id> sub_devices;
+};
+
+// (1) property name in sema_props is not a supported property name,
+// (2) value specified for a supported property name is not valid,
+// (3) the same property name is specified more than once.
+struct CreateInvalidProperty : public SemaphoreTestBase
+{
+    CreateInvalidProperty(cl_device_id device, cl_context context,
+                          cl_command_queue queue)
+        : SemaphoreTestBase(device, context, queue)
+    {}
+
+    cl_int Run() override
+    {
+        cl_int err = CL_SUCCESS;
+        // Create semaphore with invalid properties:
+        // 1) Property name in sema_props is not a supported property name
+        {
+            cl_semaphore_properties_khr sema_props[] = {
+                (cl_semaphore_properties_khr)CL_INVALID_SEMAPHORE_KHR,
+                (cl_semaphore_properties_khr)CL_SEMAPHORE_TYPE_BINARY_KHR, 0
+            };
+
+            semaphore =
+                clCreateSemaphoreWithPropertiesKHR(context, sema_props, &err);
+            test_failure_error(
+                err, CL_INVALID_PROPERTY,
+                "Unexpected clCreateSemaphoreWithPropertiesKHR return");
```

Comment:
I think there are two possible errors here:

1. There is an invalid property, which would return `CL_INVALID_PROPERTY`.
2. The properties  do not contain the required properties to create a semaphore (specifically, the semaphore type), which would return `CL_INVALID_VALUE`.

We should either allow for both error conditions, or we should disambiguate the errors so only `CL_INVALID_PROPERTY` is returned.